paint/data/dataset.py

```diff
@@ -0,0 +1,437 @@
+    item_ids : list[int]
+        List of item IDs that should be included in the dataset.
+    to_tensor : torchvision.transforms.ToTensor
+        Transform to convert image to tensor.
+
+    Methods
+    -------
+    from_benchmark()
+        Class method to initialize dataset from a benchmark file.
+    from_heliostats()
+        Class method to initialize dataset from one or more heliostats.
+    """
+
+    def __init__(
+        self,
+        root_dir: Union[str, Path],
+        item_type: str,
+        item_ids: Union[list[int], None] = None,
+    ) -> None:
+        """
+        Initialize a PaintCalibrationDataset.
+
+        This dataset contains calibration data from the paint dataset.
+
+        Parameters
+        ----------
+        root_dir : Union[str, Path]
+            Directory where the dataset will be stored.
+        item_type : str
+            Type of item being loaded, i.e. raw image, cropped image, flux image, flux centered image, or calibration
+            properties.
+        item_ids : list[int]
+            List of item IDs that should be included in the dataset. If no list is provided, all files in that folder
+            of the specified type will be loaded.
+        """
+        self._check_accepted_keys(item_type)
+        log.info(f"Initializing a data set for {item_type} calibration items...")
+        if item_type in [
+            mappings.CALIBRATION_RAW_IMAGE_KEY,
+            mappings.CALIBRATION_CROPPED_IMAGE_KEY,
+            mappings.CALIBRATION_FLUX_IMAGE_KEY,
+            mappings.CALIBRATION_FLUX_CENTERED_IMAGE_KEY,
+        ]:
+            log.info("Note that this is a dataset containing images!")
+        else:
+            log.info(
+                "Note that this is a dataset containing dictionaries of calibration properties!"
+            )
+        self.file_identifier = self._map_to_file_identifier(item_type)
+        self.root_dir = Path(root_dir)
+        if not self.root_dir.exists():
+            raise FileNotFoundError(
+                f"The root directory {self.root_dir} does not exist - you should download the dataset first!"
+            )
+
+        if item_ids is None:
+            item_ids = []
+            for item in self.root_dir.iterdir():
+                if item.is_file() and self.file_identifier in item.name:
+                    if self.file_identifier in [
+                        mappings.CALIBRATION_RAW_IMAGE_IDENTIFIER,
+                        mappings.CALIBRATION_CROPPED_IMAGE_IDENTIFIER,
+                        mappings.CALIBRATION_FLUX_IMAGE_IDENTIFIER,
+                        mappings.CALIBRATION_FLUX_CENTERED_IMAGE_IDENTIFIER,
+                    ]:
+                        item_ids.append(int(item.name.split("_")[0]))
+                    else:
+                        item_ids.append(int(item.name.split("-")[0]))
+        self.item_ids = item_ids
+        self.to_tensor = transforms.ToTensor()
+
+    def __str__(self) -> str:
```

Comment:
I guess this can be easily tested (not sure if necessary)

Reply:
I have included a test for this